### ISSUE TITLE
SCITT Plans to make a new `iss` header?

### ISSUE THREAD

**Issue by OR13** [transmute-industries/vc-cose#8]
https://github.com/transmute-industries/vc-cose/pull/6#discussion_r1037438220

**Comment by letmaik** [comment]
I'm wondering if this is an opportunity to provide stronger meaning to such an issuer header parameter, as opposed to the undefined meaning/syntax of the `iss` JWT/CWT claim.

In particular, it could be a DID-specific issuer header parameter, e.g., `did-iss` in JOSE and some integer in COSE. It would always be a DID, and if it exists in the protected header, then `kid` must be a relative DID URL.

**Comment by OR13** [comment]
@letmaik good question.

I suspect that path could be politically more contentious, since we would be defining a term that is like `iss` but expected to be some form on IRI.

I feel its better to use the existing terms, and profile them to work, rather than create new terms.

**Comment by OR13** [comment]
https://ietf-scitt.github.io/draft-birkholz-scitt-architecture/draft-birkholz-scitt-architecture.html#name-envelope-and-claim-format

```
Protected_Header = {
  1 => int               ; algorithm identifier
  3 => tstr              ; payload type
  4 => bstr              ; Key ID
  ; TBD, Labels are temporary
  391 => tstr            ; DID of Issuer // <- this guy.
  392 => tstr            ; Feed
  393 => Reg_Info        ; Registration policy info
}
```

**Comment by OR13** [comment]
https://datatracker.ietf.org/doc/html/draft-ietf-cose-cwt-claims-in-headers-01#section-3

https://www.iana.org/assignments/cwt/cwt.xhtml

There is a conflict with `iss`... which `iss` do we mean?

**Comment by letmaik** [comment]
> There is a conflict with iss... which iss do we mean?

What do you mean exactly here?

**Comment by OR13** [comment]
@letmaik I mean that in COSE, headers are identified with numbers, and `iss` is a reserved claim in a CWT, which can be promoted to a header, see the draft here: 

https://datatracker.ietf.org/doc/html/draft-ietf-cose-cwt-claims-in-headers-01#section-3

The `iss` value could also appear as a value, with an assigned value for `391`.

It should be clear when you see `iss` in a cose header, which integer that comes from... regardless of if its a CWT or a COSESign1... perhaps I am ignorant to how COSE solves for this today.

But as I understand it, it is ambiguous which integer maps to `iss` when its in a COSE protected header, as of this comment.

**Comment by tplooker** [comment]
> I'm wondering if this is an opportunity to provide stronger meaning to such an issuer header parameter, as opposed to the undefined meaning/syntax of the iss JWT/CWT claim.

I agree this is typically done in a profile atop of technologies like JWT/CWT and VC could be interpreted as a profile. Another such relevant example where additional constraining has been done is open id, id_tokens if you look at [OpenID Connect core](https://openid.net/specs/openid-connect-core-1_0.html) the `iss` value not only needs to be a string (as defined in the JWT RFC), but it must also be an https url.
